DashAI/back/plugins/utils.py

```diff
@@ -147,12 +116,91 @@ def install_plugin_from_pypi(pypi_plugin_name: str) -> None:
     RuntimeError
         If pip install command fails
     """
+    pip_action = "install" if install else "uninstall"
+    args = ["pip", pip_action, pypi_plugin_name]
+    args if install else args.append("-y")
+
     res = subprocess.run(
-        ["pip", "install", pypi_plugin_name],
+        args,
         stderr=subprocess.PIPE,
         text=True,
     )
     if res.returncode != 0:
         errors = [line for line in res.stderr.split("\n") if "ERROR" in line]
         error_string = "\n".join(errors)
         raise RuntimeError(error_string)
+
+
+def install_plugin(plugin_name: str) -> List[type]:
+    """
+    Install and register new plugins in component registry
+
+    Parameters
+    ----------
+    plugin_name : str
+        A string with the name of the plugin in pypi to install
+
+    component_registry : ComponentRegistry
+        The current app component registry
+
+    """
+    pre_installed_plugins: List[type] = get_available_plugins()
+    install_plugin_from_pypi(plugin_name)
+    installed_plugins = set(get_available_plugins()) - set(pre_installed_plugins)
+    return installed_plugins
+
+
+def register_plugin_components(installed_plugins, component_registry):
+    for plugin in installed_plugins:
+        try:
+            component_registry.register_component(plugin)
+        except Exception as e:
+            logging.exception(f"Plugin {plugin} could not be registered. Error: {e}")
```

Comment:
The same about exceptions here. Here I would simply call the `register_component` method directly and have the method throw the exception in case it detects errors.

Reply:
Fixed! To be sure, we should change that to simply call 
`
component_registry.register_component(plugin)
`
and that call should throw and catch the exception, right? (the change has been made, if you want to check how it looks like)